xbmc/utils/StringUtils.cpp

```diff
@@ -354,6 +355,18 @@ wchar_t toupperUnicode(const wchar_t& c)
   return c;
 }
 
+template <typename Type = char> struct toCapitalize
+{
+   toCapitalize() : bFirst(true) {}
+
+   Type operator()(Type c)
+   {
+      return std::isspace(c) ? (bFirst = true, c) : bFirst ? (bFirst = false, std::toupper(c)) : c;
```

Comment:
I don't understand any of that:) Can you make it more verbose so it's clearer that is changes state and what else doing on?